### PR TITLE
fix(DTFS2-5932): coverStartDate and coverEndDate fix - timezone

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Issue: 
* timezone issues meant that coverStart and coverEnd dates were the day before at 11pm due to GMT

# Changes made:
* Added helper function to set date to midnight of the date

### CODE REVIEW

File: portal-api/src/v1/helpers/convertToTimestamp.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/**
+ * @param {String} time
+ * @returns {Date} timestamp
+ * Helper function to take time as String (eg July 19, 2022)
+ * Converts to utc timestamp by getting year date and month and setting hours to midnight
+ * returns in date format
+ */
+const convertToTimestamp = (time) => {
+  const date = new Date(time);
```
Comment (abhi-markan) [inline portal-api/src/v1/helpers/convertToTimestamp.js:9]: Can we please add a check for `time` if the argument supplied is empty.

File: portal-api/src/v1/gef/models/facilities.js [hunk 2]
```diff
@@ -81,15 +82,17 @@ class Facility {
...
 
       if (req.coverEndDate != null) {
-        this.coverEndDate = new Date(req.coverEndDate);
+        const timestamp = convertToTimestamp(req.coverEndDate);
+        this.coverEndDate = new Date(timestamp);
       }
 
       if (req.issueDate != null) {
-        this.issueDate = new Date(req.issueDate);
+        this.issueDate = Date(req.issueDate);
```
Comment (abhi-markan) [inline portal-api/src/v1/gef/models/facilities.js:95]: Does this not require creating a new date class object - `new Date()` ?

File: portal-api/api-tests/helpers/convertToTimestamp.api-test.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const convertToTimestamp = require('../../src/v1/helpers/convertToTimestamp');
+
+describe('convertToTimestamp()', () => {
+  it('Should return a timestamp in the normal time format for date passed as a string', () => {
```
Comment (abhi-markan) [inline portal-api/api-tests/helpers/convertToTimestamp.api-test.js:4]: Need some more tests cases please for substanstive test coverage, could you please introduce following test scenarios:

1. Null or empty `timestamp` provided.
2. Epoch provided instead of string.

I understand helper function is being called after various UI level validation (so above arguments are edge cases) but a safety net in a helper function improves code quality and reliability.

File: portal-api/api-tests/v1/gef/facilities.api-test.js [hunk 4]
```diff
@@ -366,6 +366,39 @@ describe(baseUrl, () => {
       const { status } = await as(aMaker).put({}).to(`${baseUrl}/doesnotexist`);
       expect(status).toEqual(204);
     });
+
+    it('should update the coverStartDate and coverEndDate in the right format set to midnight', async () => {
+      const { details } = newFacility;
```
Comment (abhi-markan) [inline portal-api/api-tests/v1/gef/facilities.api-test.js:371]: Can you please add following test scenario:

1. Leap year date
2. Date with year `9999` and `10000`

File: portal-api/api-tests/v1/gef/facilities.api-test.js [hunk 5]
```diff
@@ -366,6 +366,39 @@ describe(baseUrl, () => {
...
+      const update = {
+        hasBeenIssued: true,
+        name: 'Test',
+        currency: { id: 'GBP' },
```
Comment (abhi-markan) [inline portal-api/api-tests/v1/gef/facilities.api-test.js:375]: Do we have a constant for currency `GBP`, if not then please ignore. :-)

**Review by abhi-markan — COMMENTED** [review]
Hello @Zainzzkk have left comments, could you please have a look. Thank you.

File: portal-api/src/v1/gef/models/facilities.js (hunk 2 above)
Comment (Zainzzkk) [reply portal-api/src/v1/gef/models/facilities.js:95]: fixed

File: portal-api/src/v1/helpers/convertToTimestamp.js (hunk 1 above)
Comment (Zainzzkk) [reply portal-api/src/v1/helpers/convertToTimestamp.js:9]: done

File: portal-api/api-tests/helpers/convertToTimestamp.api-test.js (hunk 3 above)
Comment (Zainzzkk) [reply portal-api/api-tests/helpers/convertToTimestamp.api-test.js:4]: done

File: portal-api/api-tests/v1/gef/facilities.api-test.js (hunk 4 above)
Comment (Zainzzkk) [reply portal-api/api-tests/v1/gef/facilities.api-test.js:371]: done

File: portal-api/api-tests/v1/gef/facilities.api-test.js (hunk 5 above)
Comment (Zainzzkk) [reply portal-api/api-tests/v1/gef/facilities.api-test.js]: done

**Review by abhi-markan — APPROVED** [review]
Good work
